plugins/services/src/js/components/forms/HealthChecksFormSection.js

```diff
@@ -0,0 +1,241 @@
+    const {errors} = this.props;
+    if (healthCheck.protocol !== 'COMMAND' && healthCheck.protocol !== 'HTTP' &&
+      healthCheck.protocol !== 'HTTPS') {
+      return null;
+    }
+
+    return (
+      <CollapsibleContainer label="Advanced Settings">
+        <div className="flex row">
+          <FormGroup
+            className="column-3"
+            required={false}
+            showError={Boolean(errors.healthChecks[key])}>
+            <FieldLabel>Grace Period (s)</FieldLabel>
+            <FieldInput
+              name={`healthChecks.${key}.gracePeriodSeconds`}
+              type="number"
+              min="0"
+              value={healthCheck.gracePeriodSeconds}/>
+            <FieldError>{errors.healthChecks[key]}</FieldError>
+          </FormGroup>
+          <FormGroup
+            className="column-3"
+            required={false}
+            showError={Boolean(errors.healthChecks[key])}>
+            <FieldLabel>Interval (s)</FieldLabel>
+            <FieldInput
+              name={`healthChecks.${key}.intervalSeconds`}
+              type="number"
+              min="0"
+              value={healthCheck.intervalSeconds}/>
+            <FieldError>{errors.healthChecks[key]}</FieldError>
+          </FormGroup>
+          <FormGroup
+            className="column-3"
+            required={false}
+            showError={Boolean(errors.healthChecks[key])}>
+            <FieldLabel>Timeout (s)</FieldLabel>
+            <FieldInput
+              name={`healthChecks.${key}.timeoutSeconds`}
+              type="number"
+              min="0"
+              value={healthCheck.timeoutSeconds}/>
+            <FieldError>{errors.healthChecks[key]}</FieldError>
+          </FormGroup>
+          <FormGroup
+            className="column-3"
+            required={false}
+            showError={Boolean(errors.healthChecks[key])}>
+            <FieldLabel>Max Failures</FieldLabel>
+            <FieldInput
+              name={`healthChecks.${key}.maxConsecutiveFailures`}
+              type="number"
+              min="0"
+              value={healthCheck.maxConsecutiveFailures}/>
+            <FieldError>{errors.healthChecks[key]}</FieldError>
+          </FormGroup>
+        </div>
+      </CollapsibleContainer>);
+  }
+
+  getCommandFields(healthCheck, key) {
+    const {errors} = this.props;
+
+    if (healthCheck.protocol !== 'COMMAND') {
+      return null;
+    }
+
+    return (
+      <div className="flex row">
+        <FormGroup
+          className="column-12"
+          required={false}
+          showError={Boolean(errors.healthChecks[key])}>
+          <FieldLabel>Command</FieldLabel>
+          <FieldTextarea
+            name={`healthChecks.${key}.command`}
+            type="text"
+            value={healthCheck.command}/>
+          <FieldError>{errors.healthChecks[key]}</FieldError>
+        </FormGroup>
+      </div>
+    );
+  }
+
+  getHTTPFields(healthCheck, key) {
+    const {errors} = this.props;
```

Comment:
Why not just const {healthChecks} = this.props.errors, since that's the only one being used?